core/space/fuse/controller.go

```diff
@@ -0,0 +1,106 @@
+package fuse
+
+import (
+	"errors"
+
+	"github.com/FleekHQ/space-poc/config"
+	fuse "github.com/FleekHQ/space-poc/core/libfuse"
+	"github.com/FleekHQ/space-poc/core/store"
+	"github.com/FleekHQ/space-poc/log"
+)
+
+// Controller is the space domain controller for managing the fuse VFS.
+// It is used by the grpc server and app/daemon generally
+type Controller struct {
+	cfg      config.Config
+	vfs      *fuse.VFS
+	store    store.Store
+	isServed bool
+}
+
+func NewFuseController(
+	cfg config.Config,
+	store store.Store,
+	vfs *fuse.VFS,
+) *Controller {
+	return &Controller{
+		cfg:      cfg,
+		store:    store,
+		vfs:      vfs,
+		isServed: false,
+	}
+}
+
+// ShouldMount check the store and config to determine if the fuse drive was previously mounted
+func (s *Controller) ShouldMount() bool {
+	mountFuseDrive, err := s.store.Get([]byte(config.MountFuseDrive))
+	if err == nil {
+		return string(mountFuseDrive) == "true"
+	} else {
+		log.Printf("Error fetching mountFuseDrive state: %s\n", err.Error())
+	}
+
+	return s.cfg.GetString(config.MountFuseDrive, "false") == "true"
+}
+
+// Mount mounts the vfs drive and immediately serves the handler.
+// It starts the Fuse Server in the background
+func (s *Controller) Mount() error {
+	if s.vfs == nil {
+		return errors.New("fuse handler not initialized")
+	}
+
+	if s.vfs.IsMounted() {
+		return nil
+	}
+
+	if err := s.vfs.Mount(
+		s.cfg.GetString(config.FuseDriveName, "FleekSpace"),
+	); err != nil {
+		return err
+	}
+
+	// persist mount state to store to trigger remount on restart
+	if err := s.store.Set([]byte(config.MountFuseDrive), []byte("true")); err != nil {
+		return err
+	}
+
+	s.serve()
+	return nil
+}
+
+func (s *Controller) serve() {
+	if s.isServed {
```

Comment:
does this need a lock or are we always calling it from one go routine?

Reply:
No, I don't think so. It is currently only called from the `Mount()` function and that now has a lock